pkg/scheduler/framework/v1alpha1/framework.go

```diff
@@ -422,57 +423,61 @@ func (f *framework) RunUnreservePlugins(
 func (f *framework) RunPermitPlugins(
 	pc *PluginContext, pod *v1.Pod, nodeName string) *Status {
 	timeout := maxTimeout
-	statusCode := Success
+	var needsWait bool
+	var waitPermitPluginName string
 	for _, pl := range f.permitPlugins {
 		status, d := pl.Permit(pc, pod, nodeName)
-		if !status.IsSuccess() {
-			if status.Code() == Unschedulable {
-				msg := fmt.Sprintf("rejected by %v at permit: %v", pl.Name(), status.Message())
-				klog.V(4).Infof(msg)
-				return NewStatus(status.Code(), msg)
-			}
-			if status.Code() == Wait {
-				// Use the minimum timeout duration.
-				if timeout > d {
-					timeout = d
-				}
-				statusCode = Wait
-			} else {
-				msg := fmt.Sprintf("error while running %v permit plugin for pod %v: %v", pl.Name(), pod.Name, status.Message())
-				klog.Error(msg)
-				return NewStatus(Error, msg)
+		switch status.Code() {
+		case Success:
+			break
+		case Unschedulable:
+			msg := fmt.Sprintf("rejected by %v at permit: %v", pl.Name(), status.Message())
+			klog.V(4).Infof(msg)
+			return NewStatus(status.Code(), msg)
+		case Wait:
+			if timeout <= d {
+				break
 			}
+			timeout = d
+			waitPermitPluginName = pl.Name()
```

Comment:
Not related much to this PR: If there are many permit plugins(e.g. permit plugin A and B) returns `Wait` with different timeout, suppose plugin A's wait timeout is smaller, need we just wait plugin A's permit or need to wait both plugin A and B's permit?

cc @bsalamat

Reply:
We need to wait for the longer timeout of plugin A and B until one of them approves the pod or denies it.